File: test_data/code/java/1005-maximize-sum-of-array-after-k-negations.java
import java.util.*;

class Solution {
    public int largestSumAfterKNegations(int[] nums, int k) {
        Arrays.sort(nums);
        int sum = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] < 0) {
                nums[i] = -nums[i];
                k--;
                if (k == 0) break;
            }
        }
        Arrays.sort(nums);
        for (int i = nums.length - 1; i >= 0; i--) {
            if (nums[i] > 0) {
                sum += nums[i];
                break;
            }
            sum += nums[i];
        }
        return sum;
    }
}

File: test_data/code/python/1005-maximize-sum-of-array-after-k-negations.py
class Solution:
    def largestSumAfterKNegations(self, nums: List[int], k: int) -> int:
        nums.sort() # Sort the array in ascending order
        count = 0 # Count of negative numbers
        for num in nums:
            if num < 0: # If number is negative
                count += 1 # Increment the count
                num = -num # Flip the number
            if count >= k: # If count of negative numbers is more than or equal to k
                break # Stop the loop
            nums[nums.index(num)] = num # Update the array with the flipped number
        nums.sort(reverse=True) # Sort the array in descending order
        return sum(nums) # Calculate and return the sum